languages/java/highlights.scm

```diff
@@ -139,11 +139,11 @@
 
 ; Annotations
 (annotation
-  "@" @attribute
+  "@" @punctuation
   name: (identifier) @attribute)
 
 (marker_annotation
-  "@" @attribute
+  "@" @punctuation
```

Comment:
I looked Zed's Rust syntax highlighting and it uses `@punctuation.special` for Rust's equivalent of Java's `@` punctuation. I see no reason not to use this somewhat more specific capture.